src/api/java/com/minecolonies/api/colony/buildings/modules/ICraftingBuildingModule.java

```diff
@@ -80,20 +81,23 @@ public interface ICraftingBuildingModule extends IBuildingModule
     String getCustomRecipeKey();
 
     /**
-     * Check if this building type can learn (or otherwise process)
-     * vanilla crafting recipes of some kind.
+     * Gets the set of recipe types that can possibly be taught to this module.
      *
-     * @return True if so; false otherwise.
+     * @return the set of recipe types that can be learned (may be empty)
      */
-    boolean canLearnCraftingRecipes();
+    @NotNull
+    Set<RecipeType<?>> getLearnableRecipeTypes();
 
     /**
      * Check if this building type can learn (or otherwise process)
-     * vanilla smelting recipes of some kind.
+     * vanilla crafting recipes of some kind.
      *
      * @return True if so; false otherwise.
      */
-    boolean canLearnFurnaceRecipes();
+    default boolean canLearnRecipeType(@NotNull RecipeType<?> type)
```

Comment:
that's pretty similar to what I did with the alchemist, but we need a custom type to include brewing.

Reply:
Well, that's annoying.  Why wouldn't minecraft/forge use their standard crafting recipe system for potion crafting?

Reply:
Because they're consistently inconsistent

Reply:
Hmm.  Reading up on potions it looks like they're not implemented in a datapack-friendly format at all.  So quite possibly you should be using an entirely separate kind of module for them, because whatever they are, it's not crafting.

Reply:
Nah, it fits very will in there.

Reply:
Welp, now I've seen the alchemist code, and this is definitely the better way to do it.  I could perhaps add a brewing recipe type to this to make it easier to port your PR to use this method instead?

Reply:
Though I wasn't originally planning to backport this to 1.16 since much of it is DO-focused, but I guess I should if you're going to want to build alchemist things on top of it.